### ISSUE TITLE
How should I set net_input_size?

### ISSUE THREAD

**Issue by tctco** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#257]
Hi! Thank you for this great work!

I'm currently trying to implement lightweight openpose to detect some small animals. Suppose I have a video of resolution 1080 x 720 but the animals are all quite tiny (eg. occupying a 60 x 60 space). How should I set the `net_input_size` and `base_height` to achieve better performance?

**Comment by Daniil-Osokin** [comment]
Hi! I believe you may train any object detector for this problem, if you need just detection (rectangle box around the animal). [CenterNet](https://github.com/xingyizhou/CenterNet) is ok to start with.

**Comment by tctco** [comment]
Hi! Unfortunately, I need pose estimation instead of object detection.

I've done some experiments on this project though. I tried to set `net_input_size` to 160 during training, reduced `sigma` to 3, decreased `stride` to 4 (also changed the output size of the net simultaneously), and set `base_height` to 664. The result looks better now.

I'm a little confused about how 'pose estimation' works since I'm not an expert in this field. I've checked the output of the training dataset and it seems that the person takes up most space in the cropped image. However, it is not true for real image/video inputs. How does the network handle such differences in the training dataset and the testing dataset?

Thanks in advance!

**Comment by Daniil-Osokin** [comment]
The idea is, that during training it is better for the network to observe objects as they will look during inference. In general, you may train and test on the images of the same size. However, during training a lot more memory is needed, so it is hard to train at inference resolution. I would suggest to select the biggest network input size for the inference, which fits into the memory, then measure object size at this resolution and use it with some padding as `net_input_size`. You may vary it to allow larger batches. Talking about the `stride`, if keypoints are easy distinguishable in the target heatmaps, the `stride` is ok. If different keypoints of the same object instance overlaps spatially due to heatmaps size, then it is better to use smaller strides (for a given `net_input_size`).

**Comment by Daniil-Osokin** [comment]
It is ok, if works for you.
